expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/extension/OperationToRequestExtension.kt

```diff
@@ -0,0 +1,151 @@
+        builder.body(this.parseRequestBody(serializer))
+    }
+
+    builder.url(
+        if (this is UrlPathTrait && this.getUrlPath().isNotBlank()) {
+            this.parseURL(serverUrl)
+        } else {
+            serverUrl
+        }
+    )
+
+    return builder.build()
+}
+
+/**
+ * Extension function to parse the URL of an operation request.
+ *
+ * This function constructs the full URL by combining the base URL with the path and query parameters
+ * from the operation request.
+ *
+ * @param base the base URL
+ * @return the constructed URL
+ * @throws MalformedURLException if the constructed URL is invalid
+ */
+@Throws(MalformedURLException::class)
+internal fun UrlPathTrait.parseURL(base: URL): URL =
+    URL(
+        StringBuilder().apply {
+            append(base.toString().trim('/'))
+
+            if (this@parseURL.getUrlPath().isNotBlank()) {
+                append(this@parseURL.getUrlPath())
+            }
+
+            if (this@parseURL is UrlQueryParamsTrait && this@parseURL.getUrlQueryParams().isNotEmpty()) {
+                append("?")
+                this@parseURL.getUrlQueryParams().forEach { (key, values) ->
+                    if (key.isBlank()) {
+                        return@forEach
+                    }
+                    values.forEach { value -> append("$key=$value&") }
+                }
+                deleteCharAt(length - 1)
+            }
+        }.toString()
+    )
+
+/**
+ * Extension function to parse the HTTP method of an operation request.
+ *
+ * This function converts the HTTP method from the operation request into the Method enum.
+ *
+ * @return the HTTP method
+ * @throws IllegalArgumentException if the HTTP method is invalid
+ */
+@Throws(IllegalArgumentException::class)
+internal fun HttpMethodTrait.parseMethod(): Method =
+    Method.valueOf(getHttpMethod().uppercase())
+
+/**
+ * Extension function to parse the media type of operation request.
+ *
+ * This function converts the content type from the operation request into the MediaType object.
+ *
+ * @return the media type
+ * @throws IllegalArgumentException if the content type is invalid
+ */
+@Throws(IllegalArgumentException::class)
+internal fun ContentTypeTrait.parseMediaType(): MediaType =
+    MediaType.parse(getContentType())
+
+/**
+ * Extension function to parse the request body of an operation request.
+ *
+ * This function serializes the request body and constructs the RequestBody object.
+ *
+ * @param serializer instance used to serialize the request body to an InputStream
+ * @return the constructed request body
+ * @throws IllegalStateException if the request body cannot be serialized
+ * @throws IOException if an I/O error occurs
+ */
+@Throws(
+    IllegalStateException::class,
+    IOException::class
+)
+internal fun OperationRequestBodyTrait<*>.parseRequestBody(
+    serializer: SerializeRequestBodyTrait
+): RequestBody {
+    require(getRequestBody() != null) { "Request body is required!" }
```

Comment:
Why is getRequestBody() nullable in an OperationRequestBodyTrait?

Reply:
Fixed.